scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,207 @@
+    parallel : bool
+        Set to ``True`` to enable parallelization; default ``False``.
+    autoscale : bool
+        Set to ``True`` to automatically perform equilibration.
+        Consider using this option if the numerical values in the
+        constraints are separated by several orders of magnitude.
+        Default: ``False``.
+    unknown_options : dict
+        Optional arguments not used by this particular solver. If
+        `unknown_options` is non-empty, a warning is issued listing all
+        unused options.
+
+    Returns
+    -------
+    sol : dict
+        A dictionary consisting of the fields:
+
+            x : 1D array
+                The values of the decision variables that minimizes the
+                objective function while satisfying the constraints.
+            fun : float
+                The optimal value of the objective function ``c @ x``.
+            slack : 1D array
+                The (nominally positive) values of the slack variables,
+                ``b_ub - A_ub @ x``.
+            con : 1D array
+                The (nominally zero) residuals of the equality constraints,
+                ``b_eq - A_eq @ x``.
+            success : bool
+                ``True`` when the algorithm succeeds in finding an optimal
+                solution.
+            status : int
+                An integer representing the exit status of the algorithm.
+
+                ``0`` : Optimization terminated successfully.
+
+                ``1`` : Iteration limit reached.
+
+                ``2`` : Problem appears to be infeasible.
+
+                ``3`` : Problem appears to be unbounded.
+
+                ``4`` : Numerical difficulties encountered.
+
+            nit : int
+                The total number of iterations performed in all phases.
+            message : str
+                A string descriptor of the exit status of the algorithm.
+
+    """
+
+    _check_unknown_options(unknown_options)
+
+    statuses = {
+        12: (1, "Iteration limit reached."),
+        9: (0, "Optimization terminated successfully."),
+        8: (3, "The problem is unbounded."),
+        7: (2, "The problem is infeasible."),
+        12: (1, "Iteration limit reached."),
+    }
+    # "Iteration limit reached.",
+    # "The problem appears infeasible, as the phase one auxiliary "
+    # "problem terminated successfully with a residual of {0:.1e}, "
+    # "greater than the tolerance {1} required for the solution to "
+    # "be considered feasible. Consider increasing the tolerance to "
+    # "be greater than {0:.1e}. If this tolerance is unnaceptably "
+    # "large, the problem is likely infeasible.",
+    # "The problem is unbounded, as the simplex algorithm found "
+    # "a basic feasible solution from which there is a direction "
+    # "with negative reduced cost in which all decision variables "
+    # "increase.",
+    # "Numerical difficulties encountered; consider trying "
+    # "method='interior-point'.",
```

Comment:
Should remove these; I think we'll just pass along the message from HiGHS

Reply:
In that case, we should probably add messages for all the possible statuses:
```
0    HighsModelStatusNOTSET,
1    HighsModelStatusLOAD_ERROR,
2    HighsModelStatusMODEL_ERROR,
3    HighsModelStatusMODEL_EMPTY,
4    HighsModelStatusPRESOLVE_ERROR,
5    HighsModelStatusSOLVE_ERROR,
6    HighsModelStatusPOSTSOLVE_ERROR,
7    HighsModelStatusPRIMAL_INFEASIBLE,
8    HighsModelStatusPRIMAL_UNBOUNDED,
9    HighsModelStatusOPTIMAL,
10    HighsModelStatusREACHED_DUAL_OBJECTIVE_VALUE_UPPER_BOUND,
11    HighsModelStatusREACHED_TIME_LIMIT,
12    HighsModelStatusREACHED_ITERATION_LIMIT,
```